dmriprep/workflows/dwi/eddy.py

```diff
@@ -0,0 +1,97 @@
+"""
+Eddy-currents and head-motion estimation/correction.
+
+.. testsetup::
+    >>> tmpdir = getfixture('tmpdir')
+    >>> tmp = tmpdir.chdir() # changing to a temporary directory
+    >>> nb.Nifti1Image(np.zeros((90, 90, 60, 6)), None, None).to_filename(
+    ...     tmpdir.join('dwi.nii.gz').strpath)
+
+"""
+from nipype.pipeline import engine as pe
+from nipype.interfaces import utility as niu
+from niworkflows.engine.workflows import LiterateWorkflow as Workflow
+
+
+def gen_eddy_textfiles(in_file, in_meta):
+    """
+    Generate the acquisition-parameters and index files for FSL ``eddy_openmp``.
+
+    Examples
+    --------
+    >>> out_acqparams, out_index = gen_eddy_textfiles(
+    ...     "dwi.nii.gz",
+    ...     {"PhaseEncodingDirection": "j-", "TotalReadoutTime": 0.005},
+    ... )
+    >>> Path(out_acqparams).read_text()
+    '0 -1 0 0.0050000'
+
+    >>> Path(out_index).read_text()
+    '1 1 1 1 1 1'
+
+    """
+    from pathlib import Path
+    import nibabel as nb
+    from sdcflows.utils.epimanip import get_trt
+    from nipype.utils.filemanip import fname_presuffix
+
+    # Generate output file name
+    out_acqparams = fname_presuffix(in_file, suffix="_acqparams.txt", use_ext=False,)
+
+    pe_dir = in_meta["PhaseEncodingDirection"]
+    fsl_pe = ["0"] * 3
+    fsl_pe["ijk".index(pe_dir[0])] = "-1" if pe_dir.endswith("-") else "1"
+
+    # Write to the acqp file
+    Path(out_acqparams).write_text(
+        f"{' '.join(fsl_pe)} {get_trt(in_meta, in_file=in_file):0.7f}"
+    )
+
+    out_index = fname_presuffix(in_file, suffix="_index.txt", use_ext=False,)
+    Path(out_index).write_text(f"{' '.join(['1'] * nb.load(in_file).shape[3])}")
+    return out_acqparams, out_index
+
+
+def init_eddy_wf(name="eddy_wf"):
+    """
+    Create a workflow for head-motion & Eddy currents distortion estimation with FSL.
+
+    Parameters
+    ----------
+    name : :obj:`str`
+        Name of workflow (default: ``eddy_wf``)
+
+    Inputs
+    ----------
+    dwi_file
+        dwi NIfTI file
+
+    Outputs
+    -------
+    out_eddy :
+        The eddy corrected diffusion image..
+
+    """
+    from nipype.interfaces.fsl import EddyCorrect
+
+    inputnode = pe.Node(niu.IdentityInterface(fields=["dwi_file"]), name="inputnode",)
+
+    outputnode = pe.Node(niu.IdentityInterface(fields=["out_eddy"]), name="outputnode",)
+
+    workflow = Workflow(name=name)
+    workflow.__desc__ = f"""\
+Geometrical distortions derived from the so-called Eddy-currents, and head-motion
+realignment parameters were estimated with the joint modeling of FSL ``eddy_correct``
+included in FSL {EddyCorrect().version} [@eddy].
+"""
+
+    eddy_correct = pe.Node(EddyCorrect(), name="eddy_correct",)
```

Comment:
I've been checking and ``EddyCorrect`` only returns the data after correction, which is a big issue. We will need the head-motion parameters, at the very least, for further processing.

Sorry to push back on this. I went to our earlier conversation to check why we pivoted that way. We (or I) thought that FSL 5 does not have ``eddy_openmp`` - so fell back to the deprecated version.

However:

```
/opt/fsl-5.0/bin/eddy_openmp
***************************************************
The following COMPULSORY options have not been set:
	--imain	File containing all the images to estimate distortions for
	--mask	Mask to indicate brain
	--index	File containing indices for all volumes in --imain into --acqp and --topup
	--acqp	File containing acquisition parameters
	--bvecs	File containing the b-vectors for all volumes in --imain
	--bvals	File containing the b-values for all volumes in --imain
	--out	Basename for output
***************************************************

Part of FSL (ID: 5.0.11)
```
At least 5.0.11 has ``eddy_openmp``.

Reply:
My mistake. Upon opening the code for `eddy_correct`, it looks like it was just a wrapper for `flirt`. Thankfully, @slimnsour  has kept the functions for writing the index and acquisition parameters file. v5.0.11 introduced the ability to do slice-to-volume motion correction. This could be useful if the dwi json includes slice timing values.

Reply:
> Sorry for the delay on this. I figured out that the issue with the tests was that there were outdated python2 print statements in code used by `eddy_correct`. I created an updated file that lets `eddy_correct` run correctly and it is used to overwrite the base FSL one through our `.docker` folder.
> 
> And as for the `--sloppy` implementation, I'm not sure what else can be done with the limited `eddy_correct` parameters to make it faster. I'd be happy to implement any solutions if you guys have any (maybe a parameter that skips `eddy_correct` altogether?)

If migrating to `eddy_openmp`, one suggestion could be decreasing the number of iterations with `niter`.

Reply:
An alternative to installing FSL 5.0.11 could be to just include the `eddy_openmp` script in the `.docker` folder (similar to how `topup` was added). But I'm not sure if there are added dependencies that are also required.

Reply:
Are we using FSL BET?

Reply:
(or does eddy_openmp use it internally?)

Reply:
Yes, the masking workflow still uses BET. I don't think `eddy_openmp` uses BET internally. It would rely on the mask that we pass it.

Reply:
Good suggestion, I've tested just using the one script and it seems to be working just fine so it's pushed in f463e665faaab7c6bd5f5ba7b3be6be0e37ddb17. I was worried that `eddy_openmp` would have dependencies that we couldn't individually account for so I initially just installed the entire FSL 5.0.11.